File: Day 03 - 21.03.2019/CSharp/Facade/Facade/HomeActionFacade.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Facade
{
    class HomeActionFacade:HomeAction
    {
        private HomeActionFacade() { }

        public static HomeActionFacade GetInstance() { return new HomeActionFacade(); }

        public void Sleep()
        {
           
             Console.WriteLine("\nYou have selected Sleep mode\n");
             Console.WriteLine("---------------------------------------");
             Console.WriteLine(CloseTheLight(false));
             Console.WriteLine(TeethBrushing(true));
             Console.WriteLine(CloseTheDoor(false));
             Console.WriteLine("---------------------------------------\n");

        }
        public void ExitTheHouse()
        {
             Console.WriteLine("\nYou have selected Exit mode from home\n");
             Console.WriteLine("---------------------------------------");
             Console.WriteLine(CloseTheDoor(false));
             Console.WriteLine(CloseTheAir(true));
             Console.WriteLine(CloseTheLight(false));
             Console.WriteLine(OrganizationToExit(true));

             Console.WriteLine("---------------------------------------\n");
        }

    }
}

File: Day 03 - 21.03.2019/Java_Fac/HomeActionFacade.java
package ClassWork;

public class HomeActionFacade extends HomeAction{

    private HomeActionFacade(){}

    public final static HomeActionFacade GetInstance(){return new HomeActionFacade();}

    public void Sleep()
    {
        System.out.println("You have selected Sleep mode");
        System.out.println("---------------------------------------");
        System.out.println(CloseTheLight(false));
        System.out.println(TeethBrushing(true));
        System.out.println(CloseTheDoor(false));
        System.out.println("---------------------------------------");

    }
    public void ExitTheHouse()
    {
        System.out.println("You have selected Exit mode from home");
        System.out.println("---------------------------------------");
        System.out.println(CloseTheDoor(false));
        System.out.println(CloseTheAir(true));
        System.out.println(CloseTheLight(false));
        System.out.println(OrganizationToExit(true));

        System.out.println("---------------------------------------");
    }

}
